### PR TITLE
feat: Added statistics recollection and visualization

### LINKED ISSUE #9: Add other measures of performance
We should consider adding more measures of performance appart from cycles. For example, the % usage of functional units may be interesting to estimate how appropriate is the "investment in hardware": a low percentage would mean an oversized hardware.

### PR DESCRIPTION
**Pull request by endes0**
With this PR, we finally can close #9!.

This PR basically add the functionality of recollecting and showing some interesting measures of the execution of a program on the superscalar and VLIW machine. It also works on batch execution mode. In fact, while I was working on testing and implementing this feature on this mode, I discovered and fixed some bugs. There are some others that need more working and should be done in another PR(ex see #84 ). 

This PR also removes Polyfill as it was breaking [JS Map object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)

### CODE REVIEW

File: src/interface/actions/stats-actions.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        value: data
+    };
+}
+
+export function clearCyclesPerReplication() {
+    return {
+        type: SET_CYCLES_PER_REPLICATION,
+        value: []
+    };
+}
```
Comment (oxcabe) [inline src/interface/actions/stats-actions.ts:56]: Add newline at the end of files!

File: src/interface/components/Superescalar/navbar/FileBarComponent.tsx [hunk 2]
```diff
@@ -45,6 +45,7 @@ class FileBarComponent extends React.Component<any, any> {
                         <Dropdown.Item eventKey="2" onClick={() => { downloadJsonFile('memory.json', this.props.memory); }}>{this.props.t('fileBar.file.downloadMemory')}</Dropdown.Item>
                         <Dropdown.Item eventKey="3" onClick={() => { this.downloadContentFile(); }}>{this.props.t('fileBar.file.downloadContent')}</Dropdown.Item>
                         <Dropdown.Item eventKey="4" onClick={() => { this.downloadCodeFile(); }}>{this.props.t('fileBar.file.downloadCode')}</Dropdown.Item>
+                        <Dropdown.Item eventKey="5" onClick={() => { downloadJsonFile('stats.json', SuperescalarIntegration.stats.exportStats()); }}>{this.props.t('fileBar.file.downloadStats')}</Dropdown.Item>
```
Comment (oxcabe) [inline src/interface/components/Superescalar/navbar/FileBarComponent.tsx:48]: These menu buttons should always download a file, even if it's empty

File: src/interface/components/VLIW/modal/BatchModalComponent.tsx [hunk 3]
```diff
@@ -42,8 +42,8 @@ class BatchModalComponent extends React.Component<any, any> {
     setOptions() {
         VLIWIntegration.setBatchMode(
             this.state.replications,
-            this.state.cacheFailPercentage,
-            this.state.cacheFailLatency
+            this.state.cacheFailLatency,
+            this.state.cacheFailPercentage
```
Comment (oxcabe) [inline src/interface/components/VLIW/modal/BatchModalComponent.tsx:46]: Here it is, the importance of [making illegal states unrepresentable](https://ybogomolov.me/making-illegal-states-unrepresentable)! (see "Opaque Types" in this article)

File: src/core/Superescalar/ReorderBuffer.ts [hunk 4]
```diff
@@ -4,6 +4,7 @@ import { Instruction } from "../Common/Instruction";
 export interface VisualReorderBufferEntry {
   instruction: {
     id: string;
+    uuid: number;
```
[the remark this answers is not in the document]
Comment (endes0) [reply src/core/Superescalar/ReorderBuffer.ts:7]: Changing it to "uid"

**Review by oxcabe — APPROVED** [review]
LGTM

**Pull request merged**